packages/js/src/generators/release-version/release-version.spec.ts

```diff
@@ -0,0 +1,148 @@
+      projects: Object.values(projectGraph.nodes), // version all projects
+      projectGraph,
+      specifier: 'major',
+      currentVersionResolver: 'disk',
+    });
+    expect(readJson(tree, 'libs/my-lib/package.json').version).toEqual('1.0.0');
+
+    await releaseVersionGenerator(tree, {
+      projects: Object.values(projectGraph.nodes), // version all projects
+      projectGraph,
+      specifier: 'minor',
+      currentVersionResolver: 'disk',
+    });
+    expect(readJson(tree, 'libs/my-lib/package.json').version).toEqual('1.1.0');
+
+    await releaseVersionGenerator(tree, {
+      projects: Object.values(projectGraph.nodes), // version all projects
+      projectGraph,
+      specifier: 'patch',
+      currentVersionResolver: 'disk',
+    });
+    expect(readJson(tree, 'libs/my-lib/package.json').version).toEqual('1.1.1');
+
+    await releaseVersionGenerator(tree, {
+      projects: Object.values(projectGraph.nodes), // version all projects
+      projectGraph,
+      specifier: '1.2.3', // exact version
+      currentVersionResolver: 'disk',
+    });
+    expect(readJson(tree, 'libs/my-lib/package.json').version).toEqual('1.2.3');
+  });
+
+  it(`should apply the updated version to the projects, including updating dependents`, async () => {
+    await releaseVersionGenerator(tree, {
+      projects: Object.values(projectGraph.nodes), // version all projects
+      projectGraph,
+      specifier: 'major',
+      currentVersionResolver: 'disk',
+    });
+
+    expect(readJson(tree, 'libs/my-lib/package.json')).toMatchInlineSnapshot(`
+      {
+        "name": "my-lib",
+        "version": "1.0.0",
+      }
+    `);
+
+    expect(
+      readJson(tree, 'libs/project-with-dependency-on-my-pkg/package.json')
+    ).toMatchInlineSnapshot(`
+      {
+        "dependencies": {
+          "my-lib": "1.0.0",
+        },
+        "name": "project-with-dependency-on-my-pkg",
+        "version": "1.0.0",
+      }
+    `);
+    expect(
+      readJson(tree, 'libs/project-with-devDependency-on-my-pkg/package.json')
+    ).toMatchInlineSnapshot(`
+      {
+        "devDependencies": {
+          "my-lib": "1.0.0",
+        },
+        "name": "project-with-devDependency-on-my-pkg",
+        "version": "1.0.0",
+      }
+    `);
+  });
+
+  describe('not all given projects have package.json files', () => {
+    beforeEach(() => {
+      tree.delete('libs/my-lib/package.json');
+    });
+
+    it(`should error with guidance when not all of the given projects are appropriate for JS versioning`, async () => {
+      await expect(
+        releaseVersionGenerator(tree, {
+          projects: Object.values(projectGraph.nodes), // version all projects
+          projectGraph,
+          specifier: 'major',
+          currentVersionResolver: 'disk',
+        })
+      ).rejects.toThrowErrorMatchingInlineSnapshot(`
+        "The project "my-lib" does not have a package.json available at libs/my-lib/package.json.
```

Comment:
Eventually this kind of error message will be more concise and link off to dedicated docs around configuring release groups